### PR TITLE
Redeyerecords parser

### PR DESCRIPTION
**Pull request by Greengrer**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    private static final String RELEASE_AND_ARTIST_SELECTOR = "DIV.productInfo H1.artist";
+    private static final String STATUS_SELECTOR = "DIV.productInfo DIV.status";
+    private static final String CATNUMBER_SELECTOR = "DIV.productInfo DIV.details>P:last-child>SPAN";
+    private static final String PRICE_DETAILS_SELECTOR = "DIV.productInfo DIV.price";
+    private static final String IMG_SELECTOR = "DIV.productInfo DIV.image>IMG";
+    private static final String IN_STOCK_MARKER = "In Stock";
+    private static final int SHOP_ID = 9;
+
+    private static final ParserConfiguration offerParserConfiguration = ParserConfiguration.builder()
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: encapsulate into private method

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class RedEyeRecordsParser extends VinylParser {
+
+    protected static final String BASE_LINK = "https://www.redeyerecords.co.uk/custom-view/page-1";
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: private

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            .catalogNumberSelector(CATNUMBER_SELECTOR)
+            .highResolutionImageSelector(IMG_SELECTOR)
+            .inStockMarkerSelector(STATUS_SELECTOR)
+            .inStockMarker(IN_STOCK_MARKER)
+            .build();
+
+    private final RedEyeRecordsDetailedParser oneOfferParser = new RedEyeRecordsDetailedParser(offerParserConfiguration);
+
+    Set<String> getAllPages() {
+        var pageLinks = getDocument(BASE_LINK)
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: Set<String>

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                .map(pageNumberOption -> pageNumberOption.attr("value"))
+                .collect(Collectors.toSet());
+        pageLinks.add(BASE_LINK);
+        return pageLinks;
+    }
+
+    Set<String> getAllOfferLinks(Set<String> pages) {
+        return pages
+                .stream()
+                .flatMap(page -> this.getDocument(page).stream())
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java:50]: w/o this

**Review by hemiSRT8 — APPROVED** [review]

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+    @Override
+    public long getShopId() {
+        return SHOP_ID;
+    }
+
+    protected List<RawOffer> getRawOffersListFromOfferLinksSet(Set<String> offerLinks) {
+        return offerLinks
+                .stream()
+                .map(offerLink -> getRawOfferFromOfferLink(offerLink))
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: method reference

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                .filter(rawOffer -> isValid(rawOffer))
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: method reference

**Review by hemiSRT8 — DISMISSED** [review]

**Review by littlepony99 — DISMISSED** [review]

**Review by littlepony99 — DISMISSED** [review]

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+    private static final String RELEASE_AND_ARTIST_SELECTOR = "DIV.productInfo H1.artist";
+    private static final String STATUS_SELECTOR = "DIV.productInfo DIV.status";
+    private static final String CATNUMBER_SELECTOR = "DIV.productInfo DIV.details>P:last-child>SPAN";
+    private static final String PRICE_DETAILS_SELECTOR = "DIV.productInfo DIV.price";
+    private static final String IMG_SELECTOR = "DIV.productInfo DIV.image>IMG";
+    private static final String IN_STOCK_MARKER = "In Stock";
+    private static final int SHOP_ID = 9;
+
+    private static final ParserConfiguration offerParserConfiguration = createParserConfiguration();
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: Please remove this field and use

> private final RedEyeRecordsDetailedParser oneOfferParser = new RedEyeRecordsDetailedParser(createParserConfiguration()); (one line)

File: src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+    private static ParserConfiguration createParserConfiguration() {
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/util/impl/RedEyeRecordsParser.java]: Please move this method at the bottom of the class

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

**Review by littlepony99 — APPROVED** [review]

**Review by TarasOrlovskyi — APPROVED** [review]

**Pull request closed without merging**